Subject: [PATCH] cli-kex: fix pointer-sign for buf_putstring

buf_putstring takes const char*; cast curve25519 pub key buffer.
diff --git a/src/cli-kex.c b/src/cli-kex.c
index 6881b36..9ff06b0 100644
--- a/src/cli-kex.c
+++ b/src/cli-kex.c
@@ -99,7 +99,7 @@ void send_msg_kexdh_init() {
 #if DROPBEAR_CURVE25519
 		case DROPBEAR_KEX_CURVE25519:
 			cli_ses.curve25519_param = gen_kexcurve25519_param();
-			buf_putstring(ses.writepayload, cli_ses.curve25519_param->pub, CURVE25519_LEN);
+			buf_putstring(ses.writepayload, (const char *)cli_ses.curve25519_param->pub, CURVE25519_LEN);
 			break;
 #endif
 #if DROPBEAR_PQHYBRID
